catalog/static/config.js

```diff
@@ -14,16 +16,45 @@ if (window.location.hostname === 'quiltdata.com') {
     // Quilt auth
     userApi: 'https://stage-auth.quiltdata.com/accounts/api-root',
     signOutUrl: 'https://stage-auth.quiltdata.com/accounts/logout?next=%2F',
-    /* Dev config
-    team: {
-      name: "TEST"
-    },
-    userApi: 'http://localhost:5002/accounts/api-root',
-    signOutUrl: 'http://localhost:5002/accounts/logout?next=%2F',
-    */
+
+    // Team feature dev
+    // team: {
+    //   id: "SuperCorp",
+    // },
+    // userApi: 'http://localhost:5002/accounts/api-root',
+    // signOutUrl: 'http://localhost:5002/accounts/logout?next=%2F',
 
     // GitHub
     // userApi: 'https://api.github.com/user',
-    // signOutUrl: '/'
+    // signOutUrl: '/',
   };
 }
+
+const mustHave = {
+  alwaysRequireAuth: 'boolean',
+  api: 'string',
+  stripeKey: 'string',
+  userApi: 'string',
+  signOutUrl: 'string',
+};
+
+// test the config object
+const keys = Object.keys(window.__CONFIG);
+// eslint has good reasons not to like for .. of, so do it the old-fashioned way:
+for (let i = 0; i < keys.length; i += 1) {
+  const k = keys[i];
+  const v = window.__CONFIG[k];
+  // eslint-disable-next-line valid-typeof
+  if (k in mustHave && typeof v !== mustHave[k]) {
+    throw new Error(`Unexpected config['${k}'}]: ${v}`);
+  }
+}
+
+// if deployment script sets an empty team.id, there is no team
```

Comment:
If we're using the deployment script, then this file isn't used... It'll be `config.js.tmpl`.